app/src/test/java/com/github/se/signify/model/user/SaveUserToFireStoreTest.kt

```diff
@@ -0,0 +1,129 @@
+import org.mockito.Mockito.`when`
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows.shadowOf
+
+@RunWith(RobolectricTestRunner::class)
+class UserRepositoryFireStoreTest {
+
+  @Mock private lateinit var mockAuth: FirebaseAuth
+  @Mock private lateinit var mockCurrentUser: FirebaseUser
+  @Mock private lateinit var mockFirestore: FirebaseFirestore
+  @Mock private lateinit var mockCollectionReference: CollectionReference
+  @Mock private lateinit var mockDocumentReference: DocumentReference
+  @Mock private lateinit var mockDocumentSnapshot: DocumentSnapshot
+  @Mock private lateinit var mockSetTask: Task<Void>
+  @Mock private lateinit var mockGetTask: Task<DocumentSnapshot>
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.openMocks(this)
+
+    if (FirebaseApp.getApps(ApplicationProvider.getApplicationContext()).isEmpty()) {
+      FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext())
+    }
+
+    // Mock FirebaseAuth and FirebaseFirestore initialization
+    mockAuth = mock(FirebaseAuth::class.java)
+    mockFirestore = mock(FirebaseFirestore::class.java)
+
+    // Mock user's authentication
+    mockCurrentUser = mock(FirebaseUser::class.java)
+    `when`(mockAuth.currentUser).thenReturn(mockCurrentUser)
+
+    // Mock Firestore collection and document reference
+    mockCollectionReference = mock(CollectionReference::class.java)
+    mockDocumentReference = mock(DocumentReference::class.java)
+
+    `when`(mockFirestore.collection(anyString())).thenReturn(mockCollectionReference)
+    `when`(mockCollectionReference.document(anyString())).thenReturn(mockDocumentReference)
+
+    // Mock Firestore document get task
+    mockGetTask = mock(Task::class.java) as Task<DocumentSnapshot>
+    `when`(mockDocumentReference.get()).thenReturn(mockGetTask)
+  }
+
+  @Test
+  fun saveUserToFireStore_shouldAddUserWhenUserDoesNotExist() {
+    // Arrange
+    `when`(mockCurrentUser.email).thenReturn("testuser@gmail.com")
+    `when`(mockCurrentUser.displayName).thenReturn("Test User")
+
+    // Simulate that the user document does not exist
+    `when`(mockGetTask.isSuccessful).thenReturn(true)
+    `when`(mockGetTask.result).thenReturn(mockDocumentSnapshot)
+    `when`(mockDocumentSnapshot.exists()).thenReturn(false) // Document does not exist
+
+    // Simulate successful set task
+    val mockSetTask = Tasks.forResult<Void>(null) // Simulate successful Firestore set task
+    `when`(mockDocumentReference.set(any(), eq(SetOptions.merge()))).thenReturn(mockSetTask)
+
+    // Act
+    saveUserToFireStore()
+
+    // Idle the main looper to process tasks
+    shadowOf(Looper.getMainLooper()).idle()
+
+    // Assert
+    `when`(mockDocumentReference.set(any(), eq(SetOptions.merge())))
+        .thenReturn(Tasks.forResult(null))
+  }
```

Comment:
The unit tests in UserRepositoryFireStoreTest validate critical parts of the logic, such as adding a user if they don’t already exist, or verifying that existing users aren't added again. The use of mock objects and Robolectric ensures that interactions with Firebase can be tested without an actual backend. It’s important to confirm these tests cover all edge cases and potential errors.